### ISSUE TITLE
[BUG] Wrong SparseTensor.dense conversion

### ISSUE THREAD

**Issue by hontrn9122** [mit-han-lab/torchsparse#292]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Given a dense torch tensor, for example:

`test = torch.rand(3,3,2)`
The result test tensor:
![image](https://github.com/mit-han-lab/torchsparse/assets/104297935/9e778aac-79c3-468f-abdb-3aff38cc44ca)

Then I try to convert this test tensor to a SparseTensor by the following code:

`sparse_data = test.to_sparse()`

`sparse_indices = sparse_data.indices().transpose(1,0).contiguous()`
`# add batch to indice, shape (Nx4)`
`sparse_indices = torch.cat((torch.zeros(sparse_indices.size(0), 1), sparse_indices), dim=1)`

`sparse_feature = scan_data.values().view(-1,1)`

`sparse_data = SparseTensor(feats=sparse_feature.cuda(), coords=sparse_indices.cuda(), spatial_range=(3,3,2))`

Then when I convert it back to its dense counterpart, the result is different from the original one:

`print(sparse_data.dense().cpu().squeeze())`
![image](https://github.com/mit-han-lab/torchsparse/assets/104297935/545389c3-f175-4f67-b5be-9c4b024ee524)
 
Is my converting code wrong or there are any problems with the SparseTensor.dense() function?

### Environment

```markdown
- GCC:
- NVCC:
- PyTorch:2.1.2
- PyTorch CUDA: 11.8
- TorchSparse: 2.1.0
```

**Comment by ZzTodd22** [comment]
"Hello! Have you figured out how to convert a dense tensor to a sparse tensor? For instance, I'm looking to convert a dense tensor with the shape [B, T, C, H, W] to a sparse tensor. Any insights or solutions would be greatly appreciated. Thank you!"

**Comment by hontrn9122** [comment]
> "Hello! Have you figured out how to convert a dense tensor to a sparse tensor? For instance, I'm looking to convert a dense tensor with the shape [B, T, C, H, W] to a sparse tensor. Any insights or solutions would be greatly appreciated. Thank you!"

I use torch.tensor.to_sparse() to transform the dense tensor to the coo sparse tensor, then I use the indices, values, and size of the created sparse tensor to create Torchsparse sparse tensor ( SparseTensor(feats=values, coords=indices, spatial_range=size) ). Remember to transform the indices as specified in the Torchsparse docs and add the batch dimension if your original tensor does not have batch dim

**Comment by ZzTodd22** [comment]
Thank you very much, I wrote a function and it worked!
```
def from_dense(x: torch.Tensor):
    """create sparse tensor fron channel last dense tensor by to_sparse
    x must be BTHWC tensor, channel last
    """
    sparse_data = x.to_sparse(x.ndim-1)
    spatial_shape = sparse_data.shape[:]
    batch_size = sparse_data.shape[0]
    sparse_indices = sparse_data.indices().transpose(1, 0).contiguous().int()

    sparse_feature = sparse_data.values()

    return SparseTensor(feats=sparse_feature.cuda(), coords=sparse_indices.cuda(), spatial_range=spatial_shape)
```

**Comment by ZzTodd22** [comment]
> > "Hello! Have you figured out how to convert a dense tensor to a sparse tensor? For instance, I'm looking to convert a dense tensor with the shape [B, T, C, H, W] to a sparse tensor. Any insights or solutions would be greatly appreciated. Thank you!"
> 
> I use torch.tensor.to_sparse() to transform the dense tensor to the coo sparse tensor, then I use the indices, values, and size of the created sparse tensor to create Torchsparse sparse tensor ( SparseTensor(feats=values, coords=indices, spatial_range=size) ). Remember to transform the indices as specified in the Torchsparse docs and add the batch dimension if your original tensor does not have batch dim
Thank you very much, I wrote a function and it worked!
```
def from_dense(x: torch.Tensor):
    """create sparse tensor fron channel last dense tensor by to_sparse
    x must be BTHWC tensor, channel last
    """
    sparse_data = x.to_sparse(x.ndim-1)
    spatial_shape = sparse_data.shape[:]
    sparse_indices = sparse_data.indices().transpose(1, 0).contiguous().int()
    sparse_feature = sparse_data.values()

    return SparseTensor(feats=sparse_feature.cuda(), coords=sparse_indices.cuda(), spatial_range=spatial_shape)
```
